src/liballoc/lib.rs

```diff
@@ -69,7 +69,10 @@
        issue_tracker_base_url = "https://github.com/rust-lang/rust/issues/",
        test(no_crate_inject, attr(allow(unused_variables), deny(warnings))))]
 #![no_std]
-#![needs_allocator]
+
+// Don't inject an allocator in stage0 since we explicitly use alloc_system, jemalloc
+// being possibly unavailable or disallowed
+#![cfg_attr(not(stage0), needs_allocator)]
```

Comment:
I tend to agree as I fail to reproduce now, but I've upgraded rustc in the meanwhile. @joerg-krause
I see you used the second commit in your tree, therefore I believe you encountered the "make install triggers stage 0" issue as well. If that's correct, does "make -d" give the reason why make starts back at stage 0? Thanks.